compiler/vm/vm.nim

```diff
@@ -1156,86 +1448,182 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
         Skipping over generic insts results in some generics failing
         Realistically we need a tyGenericAlias instead of using tyGenericInst then reasoning into it
       ]#
-      regs[ra].intVal = ord(regs[rb].node.typ.sameTypeOrNil(regs[rc].node.typ, {ExactTypeDescValues, ExactGenericParams}))
+      doAssert false
+      #regs[ra].intVal = ord(regs[rb].node.typ.sameTypeOrNil(regs[rc].node.typ, {ExactTypeDescValues, ExactGenericParams}))
       # The types should exactly match which is why we pass `{ExactTypeDescValues, ExactGenericParams}`.
       # This solves checks with generics.
     of opcXor:
-      decodeBC(rkInt)
+      decodeBC(akInt)
       regs[ra].intVal = ord(regs[rb].intVal != regs[rc].intVal)
     of opcNot:
-      decodeB(rkInt)
-      assert regs[rb].kind == rkInt
+      decodeB(akInt)
+      # TODO: proper error reporting?
```

Comment:
Strictly speaking yes, byte code issues are basically vmgen bugs and the complier (caller) and not the VM knows whether its handling should be fatal.